spring-beans/src/main/java/org/springframework/beans/factory/config/YamlProcessor.java

```diff
@@ -300,7 +300,7 @@ else if (value instanceof Collection) {
 				}
 			}
 			else {
-				result.put(key, (value != null ? value : ""));
+				result.put(key, value);
```

Comment:
This is a too simplistic view of the problem I am afraid. `Properties` does not support `null` value (per javadoc).